discovery-infra/tests/test_auth.py

```diff
@@ -0,0 +1,282 @@
+import pytest
+import random
+from contextlib import suppress
+from string import ascii_lowercase
+from typing import Optional
+from tests.base_test import BaseTest
+from tests.conftest import env_variables
+from assisted_service_client.rest import ApiException
+from test_infra import assisted_service_api, consts, utils
+
+def random_name():
+    return ''.join(random.choice(ascii_lowercase) for i in range(10))
+
+
+@pytest.fixture()
+def api_client():
+    def get_api_client(offline_token=env_variables['offline_token'], **kwargs):
+        url = env_variables['remote_service_url']
+
+        if not url:
+            url = utils.get_local_assisted_service_url(
+                utils.get_env('PROFILE'), utils.get_env('NAMESPACE'), 'assisted-service', utils.get_env('DEPLOY_TARGET'))
+        
+        return assisted_service_api.create_client(url, offline_token, **kwargs)
+
+
+    yield get_api_client
+
+@pytest.mark.skipif(not env_variables['offline_token'], reason="not cloud env")
+class TestAuth(BaseTest):
+    @pytest.fixture()
```

Comment:
Subsequent PR will unify this with fixture from BaseTest and update existing tests